app/features/notifications/notification-hooks.ts

```diff
@@ -0,0 +1,185 @@
+  T extends NotificationType = NotificationType,
+>(select: { type: T }): UseSuspenseQueryResult<Notification<T>[]> {
+  const notificationRepository = useNotificationRepository();
+  const userRef = useUserRef();
+
+  return useSuspenseQuery({
+    queryKey: [notificationsQueryKey, select.type],
+    queryFn: () =>
+      notificationRepository.list({
+        userId: userRef.current.id,
+        type: select.type,
+      }),
+    staleTime: 0,
+    refetchOnMount: 'always',
+    select: (data) =>
+      data.filter((n): n is Notification<T> => n.type === select.type),
+  });
+}
+
+export function useNotificationByTransactionId(transactionId: string) {
+  const notificationRepository = useNotificationRepository();
+
+  return useSuspenseQuery({
+    queryKey: [notificationsQueryKey, 'byTransactionId', transactionId],
+    queryFn: () => notificationRepository.getByTransactionId(transactionId),
+  });
+}
+
+export function useHasNotifications(select: {
+  type: NotificationType;
+}): UseSuspenseQueryResult<boolean> {
+  const notificationRepository = useNotificationRepository();
+  const userRef = useUserRef();
+
+  return useSuspenseQuery({
+    queryKey: [notificationsQueryKey, 'has', select.type],
+    queryFn: () =>
+      notificationRepository.hasNotifications({
+        userId: userRef.current.id,
+        type: select?.type,
+      }),
+    staleTime: Number.POSITIVE_INFINITY,
+    gcTime: Number.POSITIVE_INFINITY,
+  });
+}
+
+/**
+ * Returns a mutation that deletes a set of notifications.
+ */
+export function useDeleteNotification() {
+  const notificationRepository = useNotificationRepository();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: ({ notificationId }: { notificationId: string }) =>
+      notificationRepository.delete(notificationId),
+    retry: 3,
+    onSuccess: () => {
+      // We do not know if there are any other notifications of the same type, so we invalidate all of them.
+      queryClient.invalidateQueries({
+        queryKey: [notificationsQueryKey, 'has'],
+      });
+    },
+  });
+}
+
+/**
+ * Returns a mutation that deletes all notifications of a specific type.
+ * Checks the cache first and only makes the API call if notifications exist.
+ */
+export function useDeleteAllNotificationsByType({
+  type,
+}: { type: NotificationType }) {
+  const { data: hasNotifications } = useHasNotifications({ type });
+  const notificationRepository = useNotificationRepository();
+  const userRef = useUserRef();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: () => {
+      if (!hasNotifications) {
+        // No notifications to delete, return early
+        return Promise.resolve();
+      }
+
+      return notificationRepository.deleteAllByType({
+        userId: userRef.current.id,
+        type,
+      });
+    },
+    onSuccess: () => {
+      // We know that all notifications of this type have been deleted, so we can set the cache to false.
```

Comment:
not sure you can say so. there could have been a new one received just slightly after you deleted the transaction. if the notif for the new one created was received in `useOnNotificationCreated` before this set data to false, you would make it false even though it should be true